src/FacadeAccountCreation/FacadeAccountCreation.Core/Services/Messaging/MessagingService.cs

```diff
@@ -453,19 +452,27 @@ public class MessagingService : IMessagingService
 
         private string GetRegulatorEmail(string nation)
         {
-            nation = Regex.Replace(nation, @"\s", string.Empty);
-            switch (nation)
-            {
-                case "England":
-                    return _regulatorEmailConfig.England.ToLower();
-                case "Scotland":
-                    return _regulatorEmailConfig.Scotland.ToLower();
-                case "Wales":
-                    return _regulatorEmailConfig.Wales.ToLower();
-                case "NorthernIreland":
-                    return _regulatorEmailConfig.NorthernIreland.ToLower();
-                default:
-                    throw new ArgumentException("Nation not valid");
+            try
+            {
+                var timeout = TimeSpan.FromSeconds(timeoutInSeconds);
+                nation = Regex.Replace(nation, @"\s", string.Empty, RegexOptions.None, timeout);
+                switch (nation)
+                {
+                    case "England":
+                        return _regulatorEmailConfig.England.ToLower();
+                    case "Scotland":
+                        return _regulatorEmailConfig.Scotland.ToLower();
+                    case "Wales":
+                        return _regulatorEmailConfig.Wales.ToLower();
+                    case "NorthernIreland":
+                        return _regulatorEmailConfig.NorthernIreland.ToLower();
+                    default:
+                        throw new ArgumentException("Nation not valid");
+                }
+            }
+            catch
```

Comment:
I think this should only catch RegexMatchTimeoutException, which is the documented exception for a timeout on the match method.